pkg/utils/helper.go

```diff
@@ -121,24 +128,80 @@ func RunCommandOnHost(command string, arg ...string) (string, error) {
 	return string(out), nil
 }
 
-// RunCommandOnContainer runs a command on container system
-func RunCommandOnContainer(command string, arg ...string) (string, error) {
+// RunCommandOnContainerWithOutputStreams runs a command on container system and returns both the stdout and stderr output streams
+func RunCommandOnContainerWithOutputStreams(command string, arg ...string) (CommandOutputStreams, error) {
 	cmd := exec.Command(command, arg...)
 
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+	outputStreams := CommandOutputStreams{stdout.String(), stderr.String()}
+
+	if err != nil {
+		return outputStreams, fmt.Errorf("Fail to run command in container: %s", fmt.Sprint(err)+": "+stderr.String())
+	}
+
+	return outputStreams, nil
+}
+
+// RunCommandOnContainer  runs a command on container system and returns the stdout output stream
+func RunCommandOnContainer(command string, arg ...string) (string, error) {
+	outputStreams, err := RunCommandOnContainerWithOutputStreams(command, arg...)
+	return outputStreams.Stdout, err
+}
+
+// RunBackgroundCommand starts running a command on a container system in the background and returns its process ID
+func RunBackgroundCommand(command string, arg ...string) (int, error) {
+	cmd := exec.Command(command, arg...)
 	var out bytes.Buffer
 	var stderr bytes.Buffer
 	cmd.Stdout = &out
 	cmd.Stderr = &stderr
-	err := cmd.Run()
+	err := cmd.Start()
+	if err != nil {
+		return 0, fmt.Errorf("Fail to start background command in container: %s", fmt.Sprint(err)+": "+stderr.String())
+	}
+	return cmd.Process.Pid, nil
+}
+
+// Finds and kills a process with a given process ID
+func KillProcess(pid int) error {
+	process, err := os.FindProcess(pid)
 	if err != nil {
-		return "", fmt.Errorf("Fail to run command in container: %s", fmt.Sprint(err)+": "+stderr.String())
+		return fmt.Errorf("Failed to find process with pid %d to kill: %+v", pid, fmt.Sprint(err))

```

Comment:
use %w to wrap error:
`fmt.Errorf("find process with pid %d to kill: %w", pid, err)`